homeassistant/components/livisi/climate.py

```diff
@@ -116,6 +119,7 @@ def __init__(
         self._humidity_capability = humidity_capability
         self.aio_livisi = coordinator.aiolivisi
         self._attr_available = False
+        self._attr_name = f"{name} {room}"
```

Comment:
Please take a look at https://developers.home-assistant.io/blog/2022/07/10/entity_naming/
I think in this case the name of the entity should be only
```suggestion
        self._attr_name = name
```

And you should add `_attr_has_entity_name = True` as a class attribute

Reply:
I think the switch platform can be adjusted in the same way:
- remove `self._attr_name` (to leave default as `None`)
- set `_attr_has_entity_name = True` so that it picks up from the device name